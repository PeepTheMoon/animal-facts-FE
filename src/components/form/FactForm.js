import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addFact } from '../../actions/factsActions';

const FactForm = () => {
  const [image, setImage] = useState('');
  const [caption, setCaption] = useState('');
  const dispatch = useDispatch();

  const handleImageChange = ({ target }) => {
    setImage(target.value);
  };

  const handleCaptionChange = ({ target }) => {
    setCaption(target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    dispatch(addFact({
      image: image,
      caption: caption
    }));
  };

  return (
    <form onSubmit={handleSubmit}>

      <input type="text" placeholder="Enter image url" onChange={handleImageChange} value={image} />

      <input type="text" placeholder="Add animal fact here" onChange={handleCaptionChange} value={caption} />

      <button>Add Fact</button>

    </form>
  );
};

export default FactForm;
