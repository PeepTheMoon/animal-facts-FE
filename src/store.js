import { createStore } from 'redux';
import reducer, { initialState } from './components/App/reducers/AppReducer';

const store = createStore(
  reducer, 
  initialState,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
