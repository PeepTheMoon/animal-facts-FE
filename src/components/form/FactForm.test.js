import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../store';
import FactForm from './FactForm';

describe('FactForm component', () => {
  it('renders FactForm', () => {
    const wrapper = mount(
      <Provider store={store}>
        <FactForm />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
