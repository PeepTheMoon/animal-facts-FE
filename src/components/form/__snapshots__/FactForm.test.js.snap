// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactForm component renders FactForm 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <FactForm>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Enter image url"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Add animal fact here"
        type="text"
        value=""
      />
      <button>
        Add Fact
      </button>
    </form>
  </FactForm>
</Provider>
`;
